service/src/main/java/io/camunda/service/search/core/SearchQueryService.java

```diff
@@ -13,17 +13,19 @@
 import io.camunda.service.search.query.SearchQueryResult;
 import io.camunda.service.security.auth.Authentication;
 import io.camunda.service.transformers.ServiceTransformers;
+import io.camunda.zeebe.broker.client.api.BrokerClient;
 
 public abstract class SearchQueryService<T extends ApiServices<T>, Q extends SearchQueryBase, D>
     extends ApiServices<T> {
 
   protected final SearchClientBasedQueryExecutor executor;
 
   protected SearchQueryService(
+      final BrokerClient brokerClient,
```

Comment:
Good catch, I also moved back and forth with this in my head and at this point just figured "Well, let's add it, most services will need it anyway, there will be only very few services soon that don't need it, if any". I share your sentiment though and was very close to introducing a new interface as well. But this complicates things, as you mentioned already. It would be a bit cleaner, interface-wise. But I feel like it's over-engineering an issue that will not be relevant in a couple of days/weeks anymore.

Reply:
Another aspect to consider here is that the `SearchQueryService` is an abstract class (the `ApiServices` class is, too). We would need 3 different abstract classes as a result:
1. Only search (no broker client but search types) - no broker interaction required (e.g. search over entities we don't have commands for, like flow nodes).
2. Only broker (broker client but no search types) - no search required (e.g. job controller currently).
3. Both broker and search (broker client and search types) - most services (e.g. user task controller).

Not sure if that really helps then or if it's simpler to imply broker client necessity for all services, even if they only do search. It's simple to inject and to provide, although it might not make sense completely for them to have it.

Reply:
I agree, separating it into an interface would be just overengineering due to a naming/conceptual issue.

Maybe another alternative is to rename the `SearchQueryService` class to something that isn't `ES/OS`-specific?

Since this is more of a nitpick 🙃 now as we both agree that the current implementation is good, I'll leave it up to you to decide how to proceed.

Reply:
I think it's OK like this. The API services contain the broker client already. The search service is a specialization, it simply inherits the broker client dependency. The job service, for example, is a broker-only service, no search needed there for now. It extends the API services directly. Thus, all API services are automatically broker services, somehow.